RCAIDE/Library/Methods/Energy/Sources/Batteries/Lithium_Ion_Generic/compute_generic_li_cell_performance.py

```diff
@@ -103,65 +102,77 @@ def compute_generic_li_cell_performance(battery,state,bus,battery_discharge_flag
     # Calculate the current going into one cell  
     n_series          = battery.pack.electrical_configuration.series  
     n_parallel        = battery.pack.electrical_configuration.parallel 
-    n_total           = battery.pack.electrical_configuration.total 
+    n_total           = battery.pack.electrical_configuration.total
     
-    delta_t           = np.diff(time)
-    for t_idx in range(state.numerics.number_of_control_points):       
-        # ---------------------------------------------------------------------------------------------------
-        # Current State 
-        # ---------------------------------------------------------------------------------------------------
-        I_cell[t_idx]  = I_bat[t_idx]/n_parallel  
-        
-        # A voltage model from Chen, M. and Rincon-Mora, G. A., "Accurate Electrical Battery Model Capable of Predicting
-        # Runtime and I - V Performance" IEEE Transactions on Energy Conversion, Vol. 21, No. 2, June 2006, pp. 504-511
-        V_normalized  = (-1.031*np.exp(-35.*SOC[t_idx]) + 3.685 + 0.2156*SOC[t_idx] - 0.1178*(SOC[t_idx]**2.) + 0.3201*(SOC[t_idx]**3.))/4.1
-        V_oc[t_idx] = V_normalized * V_max
-        V_oc[t_idx][V_oc[t_idx] > V_max] = V_max
-             
-        # Voltage under load:
-        if battery_discharge_flag:
-            V_ul[t_idx]    = V_oc[t_idx]  - I_cell[t_idx]*R_0[t_idx]
-        else: 
-            V_ul[t_idx]    = V_oc[t_idx]  + I_cell[t_idx]*R_0[t_idx]
-            
-        # Compute internal resistance
-        R_bat        = -0.0169*(SOC[t_idx]**4) + 0.0418*(SOC[t_idx]**3) - 0.0273*(SOC[t_idx]**2) + 0.0069*(SOC[t_idx]) + 0.0043
-        R_0[t_idx]   = R_bat*battery_conditions.cell.resistance_growth_factor 
-        
-        # Compute Heat power generated by all cells
-        Q_heat_cell[t_idx] = (I_cell[t_idx]**2.)*R_0[t_idx]
-        Q_heat_pack[t_idx] = (I_pack[t_idx]**2.)*R_0[t_idx]
+    # ---------------------------------------------------------------------------------
+    # Examine Thermal Management System
+    # ---------------------------------------------------------------------------------
+    HAS = None  
+    for coolant_line in coolant_lines:
```

Comment:
I get the concept here, still trying to think of there is a better method to not have too many internal for loops

Reply:
Again, this for loop helps pass through it even when the coolant_lines is empty.